src/rdrand.c

```diff
@@ -28,8 +28,46 @@
 
 #include "rd.h"
 #include "rdrand.h"
+#include "rdtime.h"
+#include "tinycthread.h"
 
+/** Copied from glibc */
```

Comment:
We really don't want to use and maintain our own rand implementation, that's not an area of expertise for the librdkafka maintainers and is bound to be problematic sooner or later.

My suggestion is:
 * add a check for rand_r to configure.self and use it if available, else us standard rand() and enable.random.seed.
 * use rand_s() on win32
 * update the description for enable.random.seed and say that it is only used when rand_r() nor rand_s() is available.

Reply:
>We really don't want to use and maintain our own rand implementation, that's not an area of expertise for the librdkafka maintainers and is bound to be problematic sooner or later.

It is pretty simple and hadn't been changed since 1999, but okay will add runtime detection

> use rand_s() on win32

Actually I don't think that it should be used since

>The rand_s function writes a pseudorandom integer in the range 0 to UINT_MAX to the input pointer. The rand_s function uses the operating system to generate cryptographically secure random numbers. It does not use the seed generated by the srand function, nor does it affect the random number sequence used by rand.

So I guess it is even more heavy then rand() with global lock, and using it simply for random distribution is overkill